packages/components/atoms/f-toggle/src/components/Toggle.vue

```diff
@@ -0,0 +1,223 @@
+<template>
+    <label
+        :for="uniqueId"
+        data-test-id="toggle-component"
+        :class="[$style['c-toggle'],
+                 {
+                     [$style['c-toggle--checked']]: isChecked,
+                     [$style['c-toggle--disabled']]: $attrs.disabled,
+                     [$style['c-toggle--modeRTL']]: isModeRightToLeft
+                 }]">
+        <input
+            :id="uniqueId"
+            role="switch"
+            type="checkbox"
+            v-bind="$attrs"
+            :class="$style['c-toggle-input']"
+            :aria-checked="isChecked"
+            :aria-labelledby="ariaLabelledBy"
+            :aria-describedby="ariaDescribedBy"
+            @change="toggleOption">
+
+        <div :class="$style['c-toggle-control']">
+            <check-icon
+                v-if="isChecked"
+                :class="$style['c-toggle-icon']"
+                aria-hidden />
+        </div>
+    </label>
+</template>
+
+<script>
+import { CheckIcon } from '@justeattakeaway/pie-icons-vue';
+import { v4 as uuid } from 'uuid';
+
+export default {
+    name: 'FToggle',
+
+    components: {
+        CheckIcon
+    },
+
+    inheritAttrs: false,
+
+    props: {
+        isModeRightToLeft: {
+            type: Boolean,
+            default: false
+        },
+
+        ariaLabelledBy: {
+            type: String,
+            default: ''
+        },
+
+        ariaDescribedBy: {
+            type: String,
+            default: ''
+        }
+    },
+
+    computed: {
+        uniqueId () {
+            const idArray = ['toggle'];
+
+            if (this.$attrs.name) idArray.push(this.$attrs.name);
+            if (this.$attrs.id) idArray.push(this.$attrs.id);
+
+            idArray.push(this.uniqueIdentifierId);
+            return idArray.join('-');
+        },
+
+        isChecked () {
+            return this.$attrs.checked;
+        }
```

Comment:
I could be mistaken, does this component currently allow the consumer to start with a checked component? I can't see how it uses data-binding as there's no checked prop

Reply:
Yep if they use `f-toggle checked="true" />` the same way that they would be able to disable it.
It's the same thing we use form form filed checkbox